### ISSUE TITLE
Insert null integer PK doesn't update Id

### ISSUE THREAD

**Issue by gabriele-v** [praeclarum/sqlite-net#418]
Quote from https://www.sqlite.org/autoinc.html
> The AUTOINCREMENT keyword imposes extra CPU, memory, disk space, and disk I/O overhead and should be avoided if not strictly needed. It is usually not needed.

> On an INSERT, if the ROWID or INTEGER PRIMARY KEY column is not explicitly given a value, then it will be filled automatically with an unused integer, usually one more than the largest ROWID currently in use. This is true regardless of whether or not the AUTOINCREMENT keyword is used.

Now consider this code:
```
public class Test
        {
            [PrimaryKey]
            public int? Id { get; set; }
            [MaxLength(8)]
            public string TestText { get; set; }
        }

private void TestFunction()
        {
            var db = new SQLiteConnection("C:\\xxx.db");
            db.CreateTable<Test>();

            Test test = new Test();
            test.TestText = DateTime.Now.ToString();

            db.Insert(test);
            int id = test.Id.Value;
            db.Dispose();
        }
```

I would expect that **test.Id** should be updated with PK value, as it behave with Autoincrement field, instead it's always null.

**Comment by ColonelDuddits** [comment]
If you query the database, you will see that Id data _is_ actually incremented. The SQLite-Net wrapper, however, does not update the property on your object unless you mark the property with the `AutoIncrement` attribute.

**Comment by gabriele-v** [comment]
Yes, I haven't write it but ID field on DB it's incremented because it's handled by sqlite itself.

How could I query DB after insert if my only PK it's that ID? 
I have to make a query with a where for each column and it could teorically returns more than one record because I'm not querying any PK.. 

Also, I cannot use a select top 1 order by ID desc because sqlite will assign first free RowID, not latest + 1

**Comment by chopsueysensei** [comment]
maybe this could help?
http://alvinalexander.com/android/sqlite-autoincrement-insert-value-primary-key

**Comment by praeclarum** [comment]
I knew about ROWIDs but I haven't tried integrating them into the library. Looks like I should!
